src/Microsoft.Framework.Logging.Interfaces/Internal/LogValuesFormatter.cs

```diff
@@ -112,6 +113,44 @@ private static int FindIndexOf(string format, char ch, int startIndex, int endIn
 
         public string Format(object[] values)
         {
+            if (values != null)
+            {
+                for (int i = 0; i < values.Length; i++)
+                {
+                    var value = values[i];
+
+                    if (value == null)
+                    {
+                        continue;
+                    }
+
+                    // since 'string' implements IEnumerable, special case it
+                    if (value is string)
+                    {
+                        continue;
+                    }
+
+                    // if the value implements IEnumerable, build a comma separated string.
+                    var enumerable = value as IEnumerable;
+                    if (enumerable != null)
+                    {
+                        var enumerator = enumerable.GetEnumerator();
+                        var sb = new StringBuilder();
+                        while (enumerator.MoveNext())
+                        {
+                            if (sb.Length > 0)
+                            {
+                                sb.Append(", ");
+                            }
+
+                            sb.Append(enumerator.Current);
+                        }
+
+                        values[i] = sb.ToString();
```

Comment:
You can replace this whole block with `values[i] = string.Join(", ", enumerable.Cast<object>())`